suite-common/wallet-utils/src/fiatRatesUtils.ts

```diff
@@ -22,3 +22,21 @@ export const getFiatRateKeyFromTicker = (
 
     return getFiatRateKey(symbol, fiatCurrency, tokenAddress);
 };
+
+export const roundTimestampToNearestPastHour = (timestamp: Timestamp) => {
+    const date = new Date(timestamp * 1000);
+    date.setMinutes(0, 0, 0);
+    const roundedTimestamp = Math.floor(date.getTime() / 1000);
+
+    if (roundedTimestamp > timestamp) {
+        return roundedTimestamp - 3600;
+    }
+
+    return roundedTimestamp;
+};
```

Comment:
Since we have date-fns, it looks like a reinventing the wheel to me. But if you prefer custom function without date-fns, add unit test for it, please.
```suggestion
export const roundTimestampToNearestPastHour = (timestamp: Timestamp) =>
    getSeconds(
        roundToNearestHours(new Date(timestamp * 1000), { nearestTo: 60, roundingMethod: 'floor' }),
    );
```

Reply:
5e12947f373089e6ddd81f9431619b0d75561757

Reply:
hm, how about this?
```suggestion
const secondsInHour = 3600;

export const roundTimestampToNearestPastHour = (timestamp: Timestamp) => Math.floor(timestamp / secondsInHour) * secondsInHour;
```

Reply:
Why not this? https://date-fns.org/v3.6.0/docs/roundToNearestHours

Reply:
😄 @Nodonisko that was my first suggestion, but we have older version of date-fns with more complicated API and because this is only about rounding timestamp, I believe that `Math.floor(timestamp / secondsInHour) * secondsInHour` is really the best solution here because it's not using Date at all.

Reply:
Good point, It can be simplified - 5e12947f373089e6ddd81f9431619b0d75561757

Reply:
True it can be simplified - 2f3f50e058f90e7d01c801e3cc4a37c63e0e9902